<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Eight Queens</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="ChessBoardWidget" name="chessBoardWidget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="solutionLedsContainer" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>10</height>
       </size>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="SolutionLeds" name="solutionLeds" native="true">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_Help">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="aboutAct"/>
    <addaction name="aboutQtAct"/>
   </widget>
   <widget class="QMenu" name="menu_Tools">
    <property name="title">
     <string>&amp;Tools</string>
    </property>
    <addaction name="findSolutionAct"/>
    <addaction name="clearBoardAct"/>
    <addaction name="clearSolutionsAct"/>
    <addaction name="markBlockedFieldsAct"/>
    <addaction name="showLedsAct"/>
   </widget>
   <widget class="QMenu" name="menu_File">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="saveSolutionsAct"/>
    <addaction name="separator"/>
    <addaction name="quitAct"/>
   </widget>
   <addaction name="menu_File"/>
   <addaction name="menu_Tools"/>
   <addaction name="menu_Help"/>
  </widget>
  <widget class="QDockWidget" name="solutionsDockWidget">
   <property name="windowTitle">
    <string>Solutions</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QTableView" name="solutionsView">
       <property name="alternatingRowColors">
        <bool>true</bool>
       </property>
       <property name="selectionMode">
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="selectionBehavior">
        <enum>QAbstractItemView::SelectRows</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="boardSettingsDockWidget">
   <property name="windowTitle">
    <string>Board Settings</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Board Dimension</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="boardDimensionSpinBox">
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>100</number>
         </property>
         <property name="value">
          <number>8</number>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolsToolBar">
   <property name="windowTitle">
    <string>Tools</string>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="findSolutionAct"/>
   <addaction name="clearBoardAct"/>
  </widget>
  <action name="quitAct">
   <property name="text">
    <string>&amp;Quit</string>
   </property>
  </action>
  <action name="findSolutionAct">
   <property name="icon">
    <iconset resource="application.qrc">
     <normaloff>:/images/edit-find.png</normaloff>:/images/edit-find.png</iconset>
   </property>
   <property name="text">
    <string>Find &amp;solution</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="clearSolutionsAct">
   <property name="text">
    <string>&amp;Clear solutions</string>
   </property>
  </action>
  <action name="aboutAct">
   <property name="text">
    <string>&amp;About</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+A</string>
   </property>
  </action>
  <action name="markBlockedFieldsAct">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Mark &amp;blocked fields</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+B</string>
   </property>
  </action>
  <action name="saveSolutionsAct">
   <property name="text">
    <string>&amp;Save solutions</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="clearBoardAct">
   <property name="icon">
    <iconset resource="application.qrc">
     <normaloff>:/images/edit-clear.png</normaloff>:/images/edit-clear.png</iconset>
   </property>
   <property name="text">
    <string>Clear board</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="aboutQtAct">
   <property name="text">
    <string>About Qt</string>
   </property>
  </action>
  <action name="showLedsAct">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>&amp;Show leds</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ChessBoardWidget</class>
   <extends>QWidget</extends>
   <header>chessboardwidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SolutionLeds</class>
   <extends>QWidget</extends>
   <header>solutionleds.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="application.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>quitAct</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>271</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>findSolution(int)</slot>
 </slots>
</ui>
